tiledb/stdx/utility/to_underlying.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file to_underlying.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017-2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file contains an implementation of C++23 `std::to_underlying`.
+ *
+ * @section REFERENCE
+ *
+ * https://en.cppreference.com/w/cpp/utility/to_underlying
+ */
+
+#ifndef TILEDB_TO_UNDERLYING_H
+#define TILEDB_TO_UNDERLYING_H
+
+#include <type_traits>
+
+namespace stdx {
+
+/**
+ * Anticipatory implementation of C++23 `std::to_underlying`.
+ *
+ * Later versions will support `std::to_underlying` with header `<utility>`.
+ * This function allows a user to get the underlying type of the provided
+ * enumeration.
+ */
+template <class Enum>
+constexpr std::underlying_type_t<Enum> to_underlying(Enum e) noexcept {
+  return static_cast<std::underlying_type_t<Enum>>(e);
+}
+
+}  // namespace stdx
+
+#endif  // TILEDB_TO_UNDERLYING_H
```

Comment:
Nit: Needs a newline. Don't bother with another revision if this is the only change.